web/functions/src/index.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import * as admin from 'firebase-admin';
+ import {
+   beforeUserCreated, HttpsError,
+ } from "firebase-functions/v2/identity";
+ 
+ admin.initializeApp();
+ 
+ export const beforeCreated = beforeUserCreated((event) => {
+   const user = event.data;
+   // Only users of a specific domain can sign up.
+   if (!user?.email?.includes('@example.com')) {
```

Comment:
This includes `foo@example.com.foobar.baz.test` too

Reply:
Good catch, thanks! Updated with a more strict regex check